arches/app/datatypes/datatypes.py

```diff
@@ -2235,6 +2235,22 @@ def validate(self, value, row_number=None, source="", node=None, nodeid=None, st
                     errors.append({"type": error_type, "message": message})
         return errors
 
+    def pre_tile_save(self, tile, nodeid, **kwargs):
+        node = kwargs.pop("node", None)
+
+        if not node:
+            node = models.Node.objects.get(pk=nodeid)
+
+        if node.datatype not in ["resource-instance-list", "resource-instance"]:
+            return
+
+        relationships = tile.data[nodeid]
+        try:
+            for relationship in relationships:
+                relationship["resourceXresourceId"] = str(uuid.uuid4())
+        except:
```

Comment:
As we discussed, maybe it would be good to remove this exception and let any unexpected errors get logged.